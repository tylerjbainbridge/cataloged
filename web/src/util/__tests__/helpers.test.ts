import qs from 'query-string';
import {
  getQueryStringFromFilters,
  getFiltersFromQueryString,
} from '../helpers';

const filters = [
  { name: '-type', values: ['file'] },
  { name: 'search', value: 'random string' },
  { name: 'search', values: ['test string', 'another string'] },
  { name: 'label', value: 'label 1' },
  { name: 'label', values: ['label 2', 'label 3'] },
  { name: '-label', values: ['label 4'] },
  { name: 'is', value: 'favorited' },
  { name: '-is', value: 'not started' },
];

test('getQueryStringFromFilters', () => {
  const queryString = getQueryStringFromFilters(filters, '?itemId=123');

  expect(queryString).toMatchSnapshot();

  expect(qs.parse(queryString, { arrayFormat: 'bracket' })).toMatchSnapshot();
});

test('getFiltersFromQueryString', () => {
  const queryString = getQueryStringFromFilters(filters, '?itemId=123');

  expect(getFiltersFromQueryString(queryString)).toMatchSnapshot();
});
