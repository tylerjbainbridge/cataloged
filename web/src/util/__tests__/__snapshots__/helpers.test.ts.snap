// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFiltersFromQueryString 1`] = `
Object {
  "filters": Array [
    Object {
      "name": "-is",
      "value": "not started",
    },
    Object {
      "name": "-label",
      "value": "label 4",
    },
    Object {
      "name": "-type",
      "value": "file",
    },
    Object {
      "name": "is",
      "value": "favorited",
    },
    Object {
      "name": "label",
      "values": Array [
        "label 1",
        "label 2",
        "label 3",
      ],
    },
    Object {
      "name": "search",
      "values": Array [
        "random string",
        "test string",
        "another string",
      ],
    },
  ],
}
`;

exports[`getQueryStringFromFilters 1`] = `"-is=not%20started&-label=label%204&-type=file&is=favorited&itemId=123&label[]=label%201&label[]=label%202&label[]=label%203&search[]=random%20string&search[]=test%20string&search[]=another%20string"`;

exports[`getQueryStringFromFilters 2`] = `
Object {
  "-is": "not started",
  "-label": "label 4",
  "-type": "file",
  "is": "favorited",
  "itemId": "123",
  "label": Array [
    "label 1",
    "label 2",
    "label 3",
  ],
  "search": Array [
    "random string",
    "test string",
    "another string",
  ],
}
`;
